src/unit/engine/TypeScriptTestEngine.php

```diff
@@ -0,0 +1,32 @@
+<?php
+
+final class TypeScriptTestEngine extends ArcanistUnitTestEngine {
+  public function getEngineConfigurationName() {
+    return 'typescript';
+  }
+
+  protected function supportsRunAllTests() {
+    return true;
+  }
+
+  public function run() {
+    $time_start_seconds = microtime(true);
+    exec('./node_modules/.bin/tsc --project . --noEmit', $output, $return_var);
```

Comment:
Is there an easy way to detect this from the root of the project? I've noticed issues with how we do other linters right now (eslint probably) where it'll skip if you `arc diff` from a subdirectory